packages/react-ui-ag/src/ListingCell/ListingCell.js

```diff
@@ -0,0 +1,179 @@
+import React, { Component, createElement } from 'react'
+import PropTypes from 'prop-types'
+import themed from 'react-themed'
+import { parseArgs, randomId } from '@rentpath/react-ui-utils'
+import { Card, Text, RatingBar, Button } from '@rentpath/react-ui-core'
+import classNames from 'classnames'
+
+const multiType = PropTypes.oneOfType([
+  PropTypes.node,
+  PropTypes.func,
+  PropTypes.object,
+])
+
+@themed(/^ListingCell/, {
+  pure: true,
+})
+
+export default class ListingCell extends Component {
+
+  static propTypes = {
+    className: PropTypes.string,
+    theme: PropTypes.object,
+    viewType: PropTypes.string,
+    ctaSection: PropTypes.oneOfType([
+      multiType,
+      PropTypes.array,
+    ]),
+    onFavoriteToggle: PropTypes.func,
+    photos: PropTypes.array,
+    onCardClick: PropTypes.func,
+    ratings: multiType,
+    listingDetails: PropTypes.object,
+    onFavoriteClick: PropTypes.func,
+    RatingItem: PropTypes.func,
+  }
+
+  static defaultProps = {
+    theme: {},
+    ratings: { fillColor: '#FBB900', backgroundFillColor: '#9B9B9B' },
+  }
+
+  generateCtaButtons(props) {
+    const { ctaSection } = this.props
+    let ctaButtons
+
+    if (Array.isArray(ctaSection)) {
+      ctaButtons = ctaSection.map(cta => this.renderButton(cta, props))
+    } else {
+      ctaButtons = [this.renderButton(ctaSection, props)]
+    }
+
+    return ctaButtons
+  }
+
+  handleClick(func) {
+    return event => {
+      func()
+      event.stopPropagation()
+    }
+  }
+
+  renderButton(cta, props) {
+    const { theme, className } = this.props
+    const { type, onClick } = cta
+    return createElement(...parseArgs(cta, Button, {
+      ...props,
+      key: randomId(),
+      onClick: this.handleClick(onClick),
+      className: classNames(
+        theme.ListingCell_CTA,
+        theme[`ListingCell_CTA-${type}`],
+        className,
+      ),
+      'data-tid': `${type}-cta-button`,
+    }))
+  }
+
+  renderRatingBar(props) {
+    const {
+      theme,
+      ratings,
+      listingDetails,
+      viewType,
+      RatingItem,
+    } = this.props
+
+    const { listingId, rating, numRatings } = listingDetails
+
+    const [RatingsBar, ratingProps] = parseArgs(ratings, RatingBar)
+    return (
+      <RatingsBar
+        {...props}
+        {...ratingProps}
+        uniqueId={`${listingId}-${viewType}`}
+        score={rating}
+        label={`${numRatings}`}
+        className={theme.ListingCell_Rating}
+        RatingItem={RatingItem}
+        fillColor={ratings.fillColor}
+        backgroundFillColor={ratings.backgroundFillColor}
+      />
+    )
+  }
+
+  render() {
+    const {
+      theme,
+      viewType,
+      onCardClick,
+      listingDetails,
+      className,
```

Comment:
Would it make sense to have a favoriteItem?

Reply:
Yeah, as long as we decide for sure that we aren't going to have a react-ui-ag favoriteHeart component and just use the react-ui-core toggleButton, then yeah. I will wait to make that change until that choice is finalized